lib/src/components/NxGlobalSidebar/__tests__/NxGlobalSidebarFooter.test.tsx

```diff
@@ -4,63 +4,61 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import * as enzymeUtils from '../../../__testutils__/enzymeUtils';
-import 'jest-enzyme';
-
-import NxGlobalSidebarFooter, { NxGlobalSidebarFooterProps as Props }
-  from '../NxGlobalSidebarFooter';
-
-import NxTextLink from '../../NxTextLink/NxTextLink';
+import NxGlobalSidebarFooter, { NxGlobalSidebarFooterProps as Props } from '../NxGlobalSidebarFooter';
+import { rtlRender, rtlRenderElement } from '../../../__testutils__/rtlUtils';
 
 describe('NxGlobalSidebarFooter', function() {
   const minimalProps: Props = {},
-      getShallowComponent = enzymeUtils.getShallowComponent<Props>(NxGlobalSidebarFooter, minimalProps);
+      quickRender = rtlRender<Props>(NxGlobalSidebarFooter, minimalProps),
+      renderEl = rtlRenderElement<Props>(NxGlobalSidebarFooter, minimalProps);
+
+  it('passes additional classes specified as props to <footer>', function() {
```

Comment:
Nothing about it being a `<footer>` is being asserted here.

Looking at what roles we might want to look for, I find some interesting things in the specs.  `<footer>` has a default role of `contentinfo` unless it is a descendant of a of certain other elements, include those with `role="complementary"` (which the sidebar overall has).  In those cases, it just has a role of `generic` (like a div).  Further, a `<footer>` that is a descendant of a `complementary` element _may not_ be given the `contentinfo` role manually.  Looking at the definition of the `contentinfo` role, it does seem like a fit for our footer, and yet we are blocked from applying it by the rules in the specs.

It makes me think maybe we have the role for the sidebar itself wrong.  Maybe the sidebar should be a `group`, an extra wrapping `complementary` element should be added around the sidebar header, navigation, and extra content area (but not the footer), and the footer should be allowed to gain the `contentinfo` role that it would then naturally have.

Reply:
> Nothing about it being a `<footer>` is being asserted here.
> 
> Looking at what roles we might want to look for, I find some interesting things in the specs. `<footer>` has a default role of `contentinfo` unless it is a descendant of of certain other elements, including those with `role="complementary"` (which the sidebar overall has). In those cases, it just has a role of `generic` (like a div). Further, a `<footer>` that is a descendant of a `complementary` element _may not_ be given the `contentinfo` role manually. Looking at the definition of the `contentinfo` role, it does seem like a fit for our footer, and yet we are blocked from applying it by the rules in the specs.
> 
> It makes me think maybe we have the role for the sidebar itself wrong. Maybe the sidebar should be a `group`, an extra wrapping `complementary` element should be added around the sidebar header, navigation, and extra content area (but not the footer), and the footer should be allowed to gain the `contentinfo` role that it would then naturally have.

I think what you're saying makes a lot of sense. Wrapping the sidebar header, navigation and extra content with a `role="complementary"` div and not the footer would naturally give the footer a role of `contentInfo`, which would alleviate most of our concerns. However, this might be a bit tricky to achieve with the current implementation. 

Here's how the sidebar is structured currently:
```
    <aside className={classes} id={id} aria-label="global sidebar">
      <div className="nx-global-sidebar__header">
        <a className="nx-global-sidebar__product-info nx-global-sidebar__expanded-content"
           href={logoLink}>
          { logo }
        </a>
        { toggleButton }
      </div>
      { children }
    </aside>
```

Wrapping the header is quite easy, but the navigation section (rendered through `NxGlobalSidebarNavigation`) and the footer (rendered through `NxGlobalSidebarFooter`), both are rendered here as `children` of `NxGlobalSidebar`. The extra content area is also rendered as part of the footer.

So wrapping the header, navigation and extra content with one role and leaving out the footer, might not be possible. Thoughts?

Reply:
> So wrapping the header, navigation and extra content with one role and leaving out the footer, might not be possible. Thoughts?

Ah I see, you'll have to dig through the react children which is kind of ugly and inflexible (it requires the `NxGlobalSidebarFooter` to be a direct JSX child of the `NxGlobalSidebar` without even a fragment in between).  It's not unprecedented though, and in fact we have a utility that's perfect for this case.  See `childUtil.ts#splitOutFirst`.

Reply:
> > So wrapping the header, navigation and extra content with one role and leaving out the footer, might not be possible. Thoughts?
> 
> Ah I see, you'll have to dig through the react children which is kind of ugly and inflexible (it requires the `NxGlobalSidebarFooter` to be a direct JSX child of the `NxGlobalSidebar` without even a fragment in between). It's not unprecedented though, and in fact we have a utility that's perfect for this case. See `childUtil.ts#splitOutFirst`.

Oh yes, the `childUtil.ts#splitOutFirst` would definitely work, but one concern is that the extra content area (support, release, product and created by divs) are rendered as a part of  `NxGlobalSidebarFooter`, and as a result would be within the `<footer>`. 

I'm wondering if the extra wrapping `complementary` element should be added to only wrap the header and navigation, and leave the extra content to be rendered within the `<footer>`.

Reply:
> Oh yes, the childUtil.ts#splitOutFirst would definitely work, but one concern is that the extra content area (support, release, product and created by divs) are rendered as a part of NxGlobalSidebarFooter, and as a result would be within the <footer>.

I'm not sure what you mean.  Of these four children, I'm saying the first three should be wrapped in a complementary
![image](https://user-images.githubusercontent.com/3630091/216704848-16d6e51f-c6b9-4e1e-b321-700d72036387.png)

Reply:
> > Oh yes, the childUtil.ts#splitOutFirst would definitely work, but one concern is that the extra content area (support, release, product and created by divs) are rendered as a part of NxGlobalSidebarFooter, and as a result would be within the .
> 
> I'm not sure what you mean. Of these four children, I'm saying the first three should be wrapped in a complementary ![image](https://user-images.githubusercontent.com/3630091/216704848-16d6e51f-c6b9-4e1e-b321-700d72036387.png)

I had misunderstood what you said by extra content. I thought you were talking about the support, release text etc.
I have made the following changes:
Change sidebar role and update tests: 18cd56b8d6623037686896e014c4708233648f6a
Add test for extra content: 9b5e56aaab6af29e5a73b8eb9479e1d27dc6602d
Update documentation about NxGlobalSidebarFooter to be direct child: e5a9b960be168164c0cc447b2b850736bae5fd82